calico_node/startup/startup.go

```diff
@@ -65,32 +63,23 @@ func main() {
 	// An explicit value of true is required to wait for the datastore.
 	if os.Getenv("WAIT_FOR_DATASTORE") == "true" {
 		waitForConnection(client)
+		log.Info("Kubernetes is initialized as a Calico datastore")
 	} else {
 		message("Skipping datastore connection test")
 	}
 
-	// If this is a Kubernetes backed datastore then just make sure the
-	// datastore is initialized and then exit.  We don't need to explicitly
+	// If this is a Kubernetes backed datastore then we need to make sure the
+	// datastore is initialized.  We don't need to explicitly
 	// initialize the datastore for non-Kubernetes because the node resource
 	// management will do that for us.
 	if cfg.Spec.DatastoreType == api.Kubernetes {
 		message("Calico is using a Kubernetes datastore")
-		err = client.EnsureInitialized()
+		err := client.EnsureInitialized()
```

Comment:
Is `EnsureInitialized()` called in `watForConnection()`?  I need to double check but IIRC `startup.go` was failing if this was not called, but let me try again and see if this can indeed be removed.